### PR TITLE
[1321] Make update command return the number of updated rows.

### LINKED ISSUE #1321: [Feature Request] Make `UPDATE` return the number of updated rows
### Overview

Currently, the `UPDATE` command does not return anything to the user. It would be more useful if we returned the number of updated rows.

### Motivation

The number of updated rows is an obvious metric that users would want from this operation.

### Further details

This should be pretty straight forward to implement - we already have access to `metrics("numUpdatedRows")` inside of `UpdateCommand.scala`. Remember to add tests, please!

### PR DESCRIPTION
**Pull request by Silverlight42**
## Description

Resolves #1321.

## How was this patch tested?

Modified existing tests.

## Does this PR introduce _any_ user-facing changes?

Yes. The update command returns the following output:

image

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/sql/delta/commands/UpdateCommand.scala [hunk 1]
```diff
@@ -22,18 +22,18 @@ import org.apache.spark.sql.delta.actions.{AddCDCFile, AddFile, FileAction}
 import org.apache.spark.sql.delta.commands.cdc.CDCReader.{CDC_TYPE_COLUMN_NAME, CDC_TYPE_NOT_CDC, CDC_TYPE_UPDATE_POSTIMAGE, CDC_TYPE_UPDATE_PREIMAGE}
 import org.apache.spark.sql.delta.files.{TahoeBatchFileIndex, TahoeFileIndex}
 import org.apache.hadoop.fs.Path
-
```
Comment (scottsand-db) [inline core/src/main/scala/org/apache/spark/sql/delta/commands/UpdateCommand.scala:25]: nit: we have blank lines separating our import groups, so the above group is `other` and the below group is `org.apache.spark`. So add back the blank line please.

File: core/src/main/scala/org/apache/spark/sql/delta/commands/UpdateCommand.scala [hunk 2]
```diff
@@ -51,6 +51,10 @@ case class UpdateCommand(
     condition: Option[Expression])
   extends LeafRunnableCommand with DeltaCommand {
 
+  override val output: Seq[Attribute] = {
+    Seq(AttributeReference("num_updated_rows", LongType)())
```
Comment (scottsand-db) [inline core/src/main/scala/org/apache/spark/sql/delta/commands/UpdateCommand.scala:55]: I should've clarified this in the issue, my bad.

Can you make this `num_affected_rows` please? To be consistent with the DELETE command output.

https://github.com/delta-io/delta/blob/master/core/src/main/scala/org/apache/spark/sql/delta/commands/DeleteCommand.scala#L86

**Review by scottsand-db — CHANGES_REQUESTED** [review]
Requested 1 minor change. Otherwise LGTM. Thanks!

**Review by scottsand-db — APPROVED** [review]

**Comment by Silverlight42** [comment]
Thanks @scottsand-db!

**Pull request closed without merging**